modules/fedimint-ln-client/src/api.rs

```diff
@@ -211,46 +208,48 @@ where
     async fn get_remove_gateway_challenge(
         &self,
         gateway_id: PublicKey,
-    ) -> FederationResult<BTreeMap<PeerId, Option<sha256::Hash>>> {
-        // Only wait a second since removing a gateway is "best effort"
-        let deadline = fedimint_core::time::now() + Duration::from_secs(1);
-        let responses = self
-            .request_with_strategy(
-                ThresholdOrDeadline::<Option<sha256::Hash>>::new(
-                    self.all_peers().total(),
-                    deadline,
-                ),
-                REMOVE_GATEWAY_CHALLENGE_ENDPOINT.to_string(),
-                ApiRequestErased::new(gateway_id),
-            )
-            .await?;
-        Ok(responses)
+    ) -> BTreeMap<PeerId, Option<sha256::Hash>> {
+        let mut responses = BTreeMap::new();
+
+        for peer in self.all_peers() {
+            if let Ok(response) = self
+                // Only wait a second since removing a gateway is "best effort"
+                .request_single_peer_federation::<Option<sha256::Hash>>(
+                    Some(Duration::from_secs(1)),
+                    REMOVE_GATEWAY_CHALLENGE_ENDPOINT.to_string(),
+                    ApiRequestErased::new(gateway_id),
+                    *peer,
+                )
+                .await
+            {
+                responses.insert(*peer, response);
+            }
+        }
```

Comment:
Could be done in parallel by [creating a stream](https://docs.rs/tokio-stream/latest/tokio_stream/fn.iter.html), mapping it using [`.then(…)`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.then) putting the resulting stream into a [`FuturesUnordered`](https://docs.rs/futures/latest/futures/stream/struct.FuturesUnordered.html) and finally `collect`ing into the `BTreeMap`. That should divide total latency by the number of peers.

Since this is quite a bit of boiler-plate code I'd consider fusing `get_remove_gateway_challenge` and `remove_gateway`. That would mean only sending a single signature to each guardian instead of all the signatures for all guardians, but that's an improvement too imo. Sending a map of signatures is accidental complexity from using query strategies for something they weren't meant for imo.